### PR TITLE
Feature/reset password

### PR DESCRIPTION
**Pull request by tatianaIljina**

### CODE REVIEW

File: UI/src/app/services/interceptor.service.ts [hunk 1]
```diff
@@ -19,10 +20,16 @@ export class InterceptorService implements HttpInterceptor {
 
     return next.handle(request).pipe(
       tap(
-        event => {},
+        event => { },
         err => {
           if (err instanceof HttpErrorResponse && err.status == 401) {
-            this.router.navigate(['/login']);
+            this.activeRoute.queryParamMap.subscribe(r => {
```
Comment (DamienVStone) [inline UI/src/app/services/interceptor.service.ts]: Revert changes in interceptor file. And let's have some discussion about it.

Comment (tatianaIljina) [reply UI/src/app/services/interceptor.service.ts]: During sidenav's initialization we need to check if user was authorized to display his data. It leads to throwing error with status 401. EventEmitter reacts to the auth event in LoginService every time when user logs in and logs out, so this error is relevant for correct sidenav's work but makes router to navigate on /login page. Is it correct to throw 401 every time when user isn't authorized? Maybe is it better to separate the sidenav logic with 401 error (check current user, for example)

Comment (DamienVStone) [reply UI/src/app/services/interceptor.service.ts]: Maybe you need to separate logic of resetting password from main navigation logic. You don't need navigation when you resetting password. Just move your component on level higher than sidenav.

File: UI/src/styles.scss [hunk 2]
```diff
@@ -5,6 +5,7 @@
 body {
   margin: 0 !important;
   min-height: 50vh;
+  background-color: #303030;
```
Comment (DamienVStone) [inline UI/src/styles.scss]: For what?

Comment (tatianaIljina) [reply UI/src/styles.scss]: Background color in reset password component is different from the main background color in the project (.mat-drawer-container class is used by sidenav component). How can I fix it?
image

**Pull request merged**